### ISSUE TITLE
"project-monitoring" app reports Users.management.cattle.io "user-xxxxx" not found error

### ISSUE THREAD

**Issue by happyyangyuan** [rancher/rancher#19674]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Provision a kubernetes cluster with custom nodes.
2. Delete the rancher user who provisioned the cluster above.
3. Enable project monitoring or cluster monitoring in "tools" menu on the rancher-ui.
4. Go to Apps to check the status of the "project-monitoring".

**Result:**
![image](https://user-images.githubusercontent.com/9134975/56302720-ece05d80-616c-11e9-9f7b-d202e7165ea3.png)
And
![image](https://user-images.githubusercontent.com/9134975/56302779-084b6880-616d-11e9-9732-99371469ec0d.png)

Error log
```
2019/04/17 16:03:45 [ERROR] AppController p-9sd2d/project-monitoring [helm-controller] 
failed with : users.management.cattle.io "user-kzx5l" not found
```

**Other details that may be helpful:**
I guess the problem is caused by absence of the "createdBy" user.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.1
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM 8c16g 
- Kubernetes version (use `kubectl version`): v1.13.5

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-13T23:15:13Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 03:35:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 03:35:14 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by deniseschannon** [comment]
This is similar/almost duplicate of https://github.com/rancher/rancher/issues/18787, but related to project monitoring. 

Adding into the milestone so that it gets fixed/tested at the same time.

**Comment by deniseschannon** [comment]
@orangedeng can you make sure your fix also encompasses this use case?

**Comment by Ryanbeta** [comment]
same issue, absent user made it impossible to delete app inside rancher/rancher.
```
root@edf7cedcb56d:/var/lib/rancher# kubectl describe App app-sample -n p-l459j
Name:         app-sample
Namespace:    p-l459j
Labels:       <none>
Annotations:  field.cattle.io/creatorId=user-v724b
              lifecycle.cattle.io/create.helm-controller_c-5d6cq=true
API Version:  project.cattle.io/v3
Kind:         App
Metadata:
  Cluster Name:                   
  Creation Timestamp:             2018-09-07T03:44:58Z
  Deletion Grace Period Seconds:  0
  Deletion Timestamp:             2019-04-23T07:22:29Z
  Finalizers:
    clusterscoped.controller.cattle.io/helm-controller_c-5d6cq
  Generation:        0
  Initializers:      <nil>
  Resource Version:  38334466
  Self Link:         /apis/project.cattle.io/v3/namespaces/p-l459j/apps/app-sample
```

**Comment by loganhz** [comment]
Waiting for https://github.com/rancher/rancher/pull/19766 https://github.com/rancher/types/pull/801 https://github.com/rancher/system-charts/pull/37 merged and https://github.com/rancher/rancher/pull/19766 https://github.com/rancher/types/pull/801 backport to 2.2. branch.

**Comment by jiaqiluo** [comment]
The bug fix is validated on Rancher: master beffccda

Steps:
- as the admin user, enable the Github Auth
- log in as Github user1 to provision an ec2 cluster
- log in as a local user to provision a custom cluster 
- log in as the admin, deactivate those above two users
- as the admin, go to the default project in the ec2 cluster to enable project monitoring
- as the admin, go to the default project in the custom cluster to enable project monitoring

Result:
- the project monitoring is up
- no errors in Rancher's log
